packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -1201,12 +1201,17 @@ class SyncWatcher extends BaseWatcher {
 
   private async updateAvailableCreditMap (destinationChainId: number) {
     const destinationChain = this.chainIdToSlug(destinationChainId)
-    const bonder = this.bridge.getConfigBonderAddress(destinationChain)
+    const bonder = await this.getBonderAddress(destinationChain)
     const { availableCredit, baseAvailableCredit } = await this.calculateAvailableCredit(destinationChainId, bonder)
     this.availableCredit[destinationChain] = availableCredit
     this.baseAvailableCredit[destinationChain] = baseAvailableCredit
   }
 
+  async getBonderAddress (destinationChain: string): Promise<string> {
```

Comment:
Can this be cached as to potentially not make an on-chain call every time? Alternatively, can/should `this.bridge.getBonderAddress()` be cached (I don't see why not)?

Reply:
I don't think `signer.getAddress()` does any on-chain calls. The method is async on the ethers signer because some wallets might require async processing to retrieve the address but there's no on-chain rpc call

Reply:
Understood and agreed 👍